citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java

```diff
@@ -0,0 +1,350 @@
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void test_updateCitation_getByURIAndOwner_Invalid_URI() {
+        ICitation citation;
+        citation = new Citation();
+        citation.setTitle("title");
+        citation.setShortTitle("shortTitle");
+        citation.setDateFreetext("2019");
+        citation.setVolume("1.0");
+        ICitationConcept concept = new CitationConcept();
+        concept.setId("CON1");
+        concept.setOwner(user);
+        concept.setUri("www.google.com");
+        IConceptType type = new ConceptType();
+        type.setId("CTY1");
+        type.setOwner(user);
+        type.setName("forum");
+        type.setUri("www.test.com");
+        form.getConceptTags().iterator().next().setConceptUri("google");
+        
+        Mockito.when(conceptManager.getByUriAndOwner(Mockito.argThat(new URIMatcher()), any(IUser.class))).thenReturn(concept);
+
+        
+        Mockito.when(typeManager.getByUriAndOwner(Mockito.argThat(new URIMatcher()), any(IUser.class))).thenReturn(type);
+        helperToTest.updateCitation(citation, form, user);
+        assertTrue(equalsCitation(citation, updatedCitation));
+    }
+    
+    
+    private boolean comparePersons(Set<IPerson> a1, Set<IPerson> a2) {
+        
+        if (a1 != null && a2 != null) {
+
+            if (a1.size() != a2.size())
+                return false;
+            Iterator<IPerson> p1 = a1.iterator();
+            Iterator<IPerson> p2 = a2.iterator();
+            while (p1.hasNext() && p2.hasNext()) {
+                if (!comparePerson(p1.next(), p2.next())) {
+                    return false;
+                }
+            }
+            return true;
+        }
+        
+        return false;
+        
+    }
+    
+    
+  private boolean compareCreators(Set<ICreator> a1, Set<ICreator> a2) {
+        
+      if (a1 != null && a2 != null) {
+
+          if (a1.size() != a2.size())
+              return false;
+          Iterator<ICreator> p1 = a1.iterator();
+          Iterator<ICreator> p2 = a1.iterator();
+          while (p1.hasNext() && p2.hasNext()) {
+              if (!comparePerson(p1.next().getPerson(), p2.next().getPerson())) {
+                  return false;
+              }
+          }
+
+      }
+        
+        return false;
+        
+    }
+  
+  private boolean compareTags(Set<ICitationConceptTag> a1, Set<ICitationConceptTag> a2) {
+      
+      if (a1 != null && a2 != null) {
+
+          if (a1.size() != a2.size())
+              return false;
+          Iterator<ICitationConceptTag> p1 = a1.iterator();
+          Iterator<ICitationConceptTag> p2 = a2.iterator();
+          while (p1.hasNext() && p2.hasNext()) {
+              if (!compareConceptTag(p1.next(), p2.next())) {
+                  return false;
+              }
+          }
+      }
+        
+        return false;
+        
+    }
+  private boolean compareStrings(String s1, String s2) {
+      return s1.equals(s2);
```

Comment:
unnecessary method, just use equals where it's needed (you're just replacing one method with another).